garak/_plugins.py

```diff
@@ -58,40 +62,105 @@ def _extract_modules_klasses(base_klass):
     def _load_plugin_cache(self):
         if not os.path.exists(self._plugin_cache_filename):
             self._build_plugin_cache()
+
+        base_time = datetime.fromtimestamp(
+            os.path.getmtime(self._plugin_cache_filename), tz=timezone.utc
+        )
+        update_user_file = False
         if not os.path.exists(self._user_plugin_cache_filename):
+            update_user_file = True
```

Comment:
it looks like this is asserted before it's known whether there are any changes present (i.e. before it's known if any data will be written). should the `mkdir` on L79 etc. be skipped if there won't be changes?

Reply:
If `self._user_plugin_cache_filename` does not exist it will be written to even if there are no changes required, this conditional detects a need to `copy` the file which will be a `write`. The `mkdir` is to ensure the full path the file will be placed in exists.